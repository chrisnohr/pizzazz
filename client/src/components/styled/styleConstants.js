// colours
export const textColor = '#6C757D';
export const fadedText = '#AAB0B5';
export const borderColor = '#CED4DA';
export const white = '#FFF';
export const beige = '#E7DDCF';
export const pink = '#FFC2C0';
export const green = '#C4F2CB';
export const yellow = '#F7E9B7';
export const letterColor = '#555';
export const blue = '#BDEAFE';
// spacing
export const spacer = '3rem';
export const padding = '0.8rem 1.6rem';
export const respPad = '3vw 6vw';
export const respSpacer = '9vw';
// shadow
export const shadow = '1.8px 2.4px 1.8px rgba(0, 0, 0, 0.05)';
// text
export const respText = '4vw';
export const textSize = '1.1rem';
// border
export const border = `.5px solid ${borderColor}`;
// helpers
export const bottomSpacer = `
  margin-bottom: ${respSpacer};
  @media screen and (min-width: 501px) {
    margin-bottom: ${spacer};
  }
`;
export const centerH = `
  margin-left: auto;
  margin-right: auto;
`;
