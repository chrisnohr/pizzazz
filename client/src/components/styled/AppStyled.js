import styled from 'styled-components';
import { spacer, centerH } from './styleConstants';

const AppWrapper = styled.div`
  ${centerH};
  text-align: center;
  max-width: 500px;
  padding: 6rem 0 ${spacer} 0;
  box-sizing: border-box;
  font-family: Roboto, sans-serif;
`;

export default AppWrapper;
