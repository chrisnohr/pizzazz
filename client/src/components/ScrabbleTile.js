// @flow
import React, { PureComponent } from 'react';
import type { Node } from 'react';
import { LetterStyled, NumberSub } from './styled/ScrabbleTileStyled';
import { letterValues } from '../Constants';

type TileProps = {
  isValid: boolean,
  letter: string,
};

class ScrabbleTile extends PureComponent<TileProps> {
  render(): Node {
    const { letter, isValid } = this.props;

    return (
      <LetterStyled isValid={isValid} letter={letter}>
        { letter.toUpperCase() }
        <NumberSub>
          { letterValues[letter] }
        </NumberSub>
      </LetterStyled>
    );
  }
}

export default ScrabbleTile;
