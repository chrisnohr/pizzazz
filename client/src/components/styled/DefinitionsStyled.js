import styled from 'styled-components';
import * as style from './styleConstants';

const DefinitionsStyled = styled.div`
  ${style.halfBottomSpacer};
  text-align: left;
  font-style: italic;
  font-weight: 200;
  font-size: ${style.respText};
  color: ${style.textColor};
  border-left:${style.border};
  padding: ${style.respPad};
  @media screen and (min-width: 445px) {
    padding: ${style.padding};
    font-size: ${style.textSize};
  };
`;

export default DefinitionsStyled;
