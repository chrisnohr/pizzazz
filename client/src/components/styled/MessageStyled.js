import styled from 'styled-components';
import {
  textColor, borderColor, padding,
  respText, respPad, bottomSpacer,
} from './styleConstants';

export const MessageContainer = styled.div`
  ${bottomSpacer}
`;

export const MessageStyled = styled.span`
  font-size: ${respText};
  font-weight: 200;
  color: ${textColor};
  padding: ${respPad};
  border-bottom: .5px solid ${borderColor};
  @media screen and (min-width: 445px) {
    font-size: 1.1rem;
    padding: ${padding};
  };
`;
