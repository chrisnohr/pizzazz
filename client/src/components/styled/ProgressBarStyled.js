import styled from 'styled-components';
import { gameDuration } from '../../Constants';
import {
  white, beige, pink, spacer, shadow,
} from './styleConstants';

export const Container = styled.div`
  position: relative;
  height: 11px;
  background: ${white};
  padding: 4.5px;
  max-width: 500px;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: ${spacer};
  box-shadow: ${shadow};
`;

export const Bar = styled.div`
  position: relative;
  background: ${beige};
  height: 100%;
  border-right: solid ${pink} 1px;
  animation-name: status-bar;
  animation-timing-function: linear;
  animation-duration: ${props => props.duration}s;
  @keyframes status-bar {
    from { width: ${props => 100 - (props.duration / gameDuration) * 100}%; }
    to { width: 100%; }
  }
`;
