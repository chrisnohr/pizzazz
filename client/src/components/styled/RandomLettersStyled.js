import styled from 'styled-components';
import { halfBottomSpacer, centerH } from './styleConstants';

const RandomLettersStyled = styled.div`
  ${halfBottomSpacer};
  ${centerH};
  padding: 4.1px;
  list-style: none;
  display: flex;
  justify-content: space-around;
  box-sizing: border-box;
`;

export default RandomLettersStyled;
