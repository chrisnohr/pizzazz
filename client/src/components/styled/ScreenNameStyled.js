import styled from 'styled-components';
import * as style from './styleConstants';

export const FormContainer = styled.div`
  display: flex;
  justify-content: center;
  box-sizing: border-box;
`;

export const Form = styled.form`
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
`;

export const HiddenLabel = styled.label`
  display: none!important;
`;

export const FormGroup = styled.div`
  flex: 0 0 auto;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-box-flex: 0;
`;

export const InputStyled = styled.input`
  display: inline-block;
  vertical-align: middle;
  font-size: ${style.respText};
  font-weight: 200;
  padding: 3vw 6vw;
  margin-bottom: ${style.respSpacer};
  color: ${style.textColor};
  border: .5px solid ${style.borderColor};
  background-color: ${style.white};
  box-shadow: ${style.shadow};
  @media screen and (min-width: 445px) {
    margin-bottom: ${style.spacer};
    font-size: 1.1rem;
    padding: ${style.padding};
  };
  :focus {
    outline: 0;
    border: 1px solid ${style.blue};
  }
`;

export const ButtonStyled = styled(InputStyled)`
  border-radius: .3rem;
  text-align: center;
  :active {
    box-shadow: inset ${style.shadow};
  }
`;

export const ButtonForm = styled(ButtonStyled)`
  border-radius: 0 .3rem .3rem 0;
`;

export const TextInput = styled(InputStyled)`
  text-align: left;
  border-radius: .3rem 0 0 .3rem;
  ::placeholder {
    color: ${style.textColor};
  }
  :focus {
    ::placeholder {
      color: ${style.fadedText};
    }
  }
`;
