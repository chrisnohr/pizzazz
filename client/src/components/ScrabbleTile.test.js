import React from 'react';
import { shallow } from 'enzyme';
import ScrabbleTile from './ScrabbleTile';
import { LetterStyled, NumberSub } from './styled/ScrabbleTileStyled';
import { letterValues } from '../Constants'

describe('<ScrabbleTile />', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<ScrabbleTile isValid={false} letter="e" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders the letter in uppercase', () => {
    const wrapper = shallow(<ScrabbleTile isValid={false} letter="e" />);
    expect(wrapper.find(LetterStyled).children().first().text()).toEqual('E');
  });
  it('renders the right letter value for e', () => {
    const wrapper = shallow(<ScrabbleTile isValid={false} letter="e" />);
    expect(wrapper.find(NumberSub).children().first().text()).toEqual(`${letterValues.e}`);
  });
  it('renders the right letter value for z', () => {
    const wrapper = shallow(<ScrabbleTile isValid={false} letter="z" />);
    expect(wrapper.find(NumberSub).children().first().text()).toEqual(`${letterValues.z}`)
  });
});
