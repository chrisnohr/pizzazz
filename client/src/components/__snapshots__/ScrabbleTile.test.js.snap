// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrabbleTile /> matches the snapshot 1`] = `
<styled.div
  isValid={false}
  letter="e"
>
  E
  <styled.sub>
    1
  </styled.sub>
</styled.div>
`;
